frontend/public/components/factory/details.tsx

```diff
@@ -25,30 +25,35 @@ import { breadcrumbsForDetailsPage } from '../utils/breadcrumbs';
 
 export const DetailsPage = withFallback<DetailsPageProps>(({ pages = [], ...props }) => {
   const resourceKeys = _.map(props.resources, 'prop');
+
   const resourcePageExtensions = useExtensions<ResourceTabPage>(isResourceTabPage);
-  let allPages = [
-    ...pages,
-    ...resourcePageExtensions
-      .filter(
-        (p) =>
-          referenceForModel(p.properties.model) ===
-          (props.kindObj ? referenceFor(props.kindObj) : props.kind),
-      )
-      .map((p) => ({
-        href: p.properties.href,
-        name: p.properties.name,
-        component: () => (
-          <AsyncComponent
-            loader={p.properties.loader}
-            namespace={props.namespace}
-            name={props.name}
-            kind={props.kind}
-            match={props.match}
-          />
-        ),
-      })),
-  ];
+  const pluginPages = React.useMemo(
+    () =>
+      resourcePageExtensions
+        .filter(
+          (p) =>
+            referenceForModel(p.properties.model) ===
+            (props.kindObj ? referenceFor(props.kindObj) : props.kind),
+        )
+        .map((p) => ({
+          href: p.properties.href,
+          name: p.properties.name,
+          component: () => (
+            <AsyncComponent
+              loader={p.properties.loader}
+              namespace={props.namespace}
+              name={props.name}
+              kind={props.kind}
+              match={props.match}
+            />
+          ),
+        })),
+    [resourcePageExtensions, props.kind, props.kindObj, props.name, props.namespace, props.match],
```

Comment:
Note, we could also replace `props.xxx` with just `props` but that isn't 100% correct from hook dependency point of view.

The main reason for not destructuring `props` into smaller pieces is to minimize changes.